tasks/ta_finisher.py

```diff
@@ -121,64 +120,42 @@ def get_totals(
     return totals
 
 
-def populate_failures(
-    failures: list[TestResultsNotificationFailure],
-    db_session: Session,
-    repoid: int,
-    commitid: str,
-    shorten_paths: bool,
-    uploads: dict[int, Upload],
-    escaper: StringEscaper,
-) -> None:
-    failed_test_instances = latest_failures_for_commit(db_session, repoid, commitid)
-
-    for test_instance in failed_test_instances:
-        failure_message = test_instance.failure_message
-        if failure_message is not None:
-            if shorten_paths:
-                failure_message = shorten_file_paths(failure_message)
-            failure_message = escaper.replace(failure_message)
-
-        upload = uploads[test_instance.upload_id]
-
-        failures.append(
-            TestResultsNotificationFailure(
-                display_name=test_instance.test.computed_name
-                if test_instance.test.computed_name is not None
-                else test_instance.test.name,
-                failure_message=failure_message,
-                test_id=test_instance.test_id,
-                envs=upload.flag_names,
-                duration_seconds=test_instance.duration_seconds,
-                build_url=upload.build_url,
-            )
+def get_bigquery_test_data(
+    repo: Repository, commit_sha: str, commit_yaml: UserYaml
+) -> tuple[
+    PRCommentAggResult,
+    list[PRCommentFailResult[tuple[bytes, bytes | None]]],
+    dict[tuple[bytes, bytes | None], FlakeInfo] | None,
+]:
+    driver = BQDriver(repo.repoid)
+    agg_result = driver.pr_comment_agg(commit_sha)
+    failures = driver.pr_comment_fail(commit_sha)
+    if should_do_flaky_detection(repo, commit_yaml):
+        flaky_tests = driver.get_repo_flakes(
+            tuple(failure["id"] for failure in failures)
         )
+    else:
+        flaky_tests = None
 
+    return agg_result, failures, flaky_tests
 
-def get_flaky_tests(
-    db_session: Session,
-    repoid: int,
-    failures: list[TestResultsNotificationFailure],
-) -> dict[str, FlakeInfo]:
-    failure_test_ids = [failure.test_id for failure in failures]
-
-    matching_flakes = list(
-        db_session.query(Flake)
-        .filter(
-            Flake.repoid == repoid,
-            Flake.testid.in_(failure_test_ids),
-            Flake.end_date.is_(None),
-            Flake.count != (Flake.recent_passes_count + Flake.fail_count),
+
+def get_postgres_test_data(
+    db_session: Session, repo: Repository, commit_sha: str, commit_yaml: UserYaml
```

Comment:
the `repo` and `commit_yaml` are only used to initialize the driver, and for the `should_do_flaky_detection` check.
instead, you could pass the driver and that flag from the outside. That way you only need a single function instead of two.